packages/server/utils/AzureDevOpsServerManager.ts

```diff
@@ -0,0 +1,457 @@
+import {authorizeOAuth2} from '../integrations/helpers/authorizeOAuth2'
+import {
+  OAuth2PkceAuthorizationParams,
+  OAuth2PkceRefreshAuthorizationParams
+} from '../integrations/OAuth2Manager'
+import {IntegrationProviderAzureDevOps} from '../postgres/queries/getIntegrationProvidersByIds';
+import {IGetTeamMemberIntegrationAuthQueryResult} from '../postgres/queries/generated/getTeamMemberIntegrationAuthQuery';
+import {isError} from 'util';
+
+export interface AzureDevOpsUser {
+  // self: string
+  displayName: string
+  publicAlias: string
+  emailAddress: string
+  id: string
+  coreRevision: number
+  revision: number
+  timeStamp: string
+}
+
+export interface Resource {
+  accountId: string
+  accountUri: string
+  accountName: string
+  properties: any
+}
+
+export interface AccessibleResources {
+  count: number
+  value: Resource[]
+}
+
+export interface WorkItemQueryResult {
+  asOf: string
+  columns: WorkItemFieldReference[]
+  queryResultType: QueryResultType
+  queryType: QueryType
+  sortColumns: WorkItemQuerySortColumn[]
+  workItemRelations: WorkItemLink[]
+  workItems: WorkItemReference[]
+}
+
+export interface WorkItemFieldReference {
+  name: string
+  referenceName: string
+  url: string
+}
+
+export enum QueryResultType {
+  WORKITEM = 'workItem',
+  WORKITEMLINK = 'workItemLink'
+}
+
+export enum QueryType {
+  FLAT = 'flat',
+  ONEHOP = 'oneHop',
+  TREE = 'tree'
+}
+
+export interface WorkItemQuerySortColumn {
+  descending: boolean
+  field: WorkItemFieldReference
+}
+
+export interface WorkItemLink {
+  rel: string
+  source: WorkItemReference
+  target: WorkItemReference
+}
+
+export interface WorkItemReference {
+  id: number
+  url: string
+}
+
+export interface WorkItemBatchResponse {
+  count: number
+  value: WorkItem[]
+}
+
+export interface WorkItem {
+  _links: ReferenceLinks
+  commentVersionRef: WorkItemCommentVersionRef
+  fields: object
+  id: number
+  relations: WorkItemRelations[]
+  rev: number
+  url: string
+}
+
+export interface ReferenceLinks {
+  links: object
+}
+
+export interface WorkItemCommentVersionRef {
+  commentId: number
+  createdInRevision: number
+  isDeleted: boolean
+  text: string
+  url: string
+  version: number
+}
+
+export interface WorkItemRelations {
+  attributes: object
+  rel: string
+  url: string
+}
+
+export interface TeamProjectReference {
+  abbreviation: string
+  defaultTeamImageUrl: string
+  description: string
+  id: string
+  lastUpdateTime: string
+  name: string
+  revision: number
+  state: string
+  url: string
+  visibility: string
+}
+
+export interface ProjectState {
+  all: string
+  createPending: string
+  deleted: string
+  deleting: string
+  new: string
+  unchanged: string
+  wellFormed: string
+}
+
+export interface ProjectVisibility {
+  private: string
+  public: string
+}
+
+export interface AzureDevOpsError {
+  code: number
+  message: string
+}
+
+const MAX_REQUEST_TIME = 5000
+
+class AzureDevOpsServerManager {
+  fetch = fetch as any
```

Comment:
-1 no need for this property anymore, just use `fetch` from `node-fetch` directly. Also it was never set.

Reply:
The standard fetch will complain about the signal parameter since it is not cast as any.

Reply:
I removed it and tested with the node-fetch and it was failing

Reply:
-1 I checked and saw there is a type mismatch between node-fetch and the `AbortController` and `RequestInit` coming from DOM. Adding the imports
```
import fetch, {RequestInit} from 'node-fetch'
import AbortController from 'abort-controller'
```
fixes these. Then we can get rid of this member and use `fetch` directly.